lib/state/wallet/state.dart

```diff
@@ -620,4 +640,10 @@ class WalletState with ChangeNotifier {
 
     notifyListeners();
   }
+
+  void setWalletActionsLoading(bool loading) {
```

Comment:
As mentioned above, you can probably modify this. 

The pattern for loading stuff is usually:
- `walletActionsRequest();` >> set loading = true
- `walletActionsSuccess(List<ActionButton>);` >> set loading = false, walletActions = ...
- `walletActionsFail()` >> set loading = false

Since you are making a network request to check the minter role, you might need a try/catch